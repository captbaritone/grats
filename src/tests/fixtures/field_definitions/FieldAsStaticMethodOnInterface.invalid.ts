/** @gqlType */
export interface User {
  /** @gqlField */
  name: string;

  /** @gqlField */
  static getUser(_: Query): User;
}

/** @gqlType */
type Query = unknown;
