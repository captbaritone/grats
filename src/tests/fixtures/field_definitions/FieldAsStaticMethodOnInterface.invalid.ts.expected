-----------------
INPUT
----------------- 
/** @gqlType */
export interface User {
  /** @gqlField */
  name: string;

  /** @gqlField */
  static getUser(_: Query): User;
}

/** @gqlType */
type Query = unknown;

-----------------
OUTPUT
-----------------
src/tests/fixtures/field_definitions/FieldAsStaticMethodOnInterface.invalid.ts:7:3 - error: Unexpected `static` modifier on non-method `@gqlField`. `static` is only valid on method signatures.

7   static getUser(_: Query): User;
    ~~~~~~
